components/linking_landing_page/linking_landing_page.tsx

```diff
@@ -0,0 +1,354 @@
+import mobileImg from 'images/deep-linking/deeplinking-mobile-img.png';
+import MattermostLogoSvg from 'images/logo.svg';
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import CheckboxCheckedIcon from 'components/widgets/icons/checkbox_checked_icon';
+import BrowserStore from 'stores/browser_store';
+import {VaultPreferenceTypes} from 'utils/constants';
+
+import * as UserAgent from 'utils/user_agent';
+
+type Props = {
+    backgroundColor: string;
+    location: string;
+    nativeLocation: string;
+    desktopAppLink?: string;
+    iosAppLink?: string;
+    androidAppLink?: string;
+    siteUrl?: string;
+}
+
+type State = {
+    protocolUnsupported: boolean;
+    browserUnsupported: boolean;
+    rememberChecked: boolean;
+    redirectPage: boolean;
+}
+
+export default class LinkingLandingPage extends PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            protocolUnsupported: false,
+            browserUnsupported: false,
+            rememberChecked: false,
+            redirectPage: false,
+        };
+
+        this.checkVaultPreference();
+    }
+
+    checkVaultPreference = () => {
+        const vaultPreference = BrowserStore.getVaultPreference(this.props.siteUrl);
+        if (vaultPreference) {
+            switch (vaultPreference) {
+            case VaultPreferenceTypes.MATTERMOSTAPP:
+                this.openMattermostApp();
+                break;
+            case VaultPreferenceTypes.BROWSER:
+                this.openInBrowser();
+                break;
+            default:
+                break;
+            }
+        }
+    }
+
+    handleChecked = () => {
+        this.setState({rememberChecked: !this.state.rememberChecked});
+    }
+
+    setPreference = (pref: string) => {
+        if (this.state.rememberChecked) {
+            switch (pref) {
+            case VaultPreferenceTypes.MATTERMOSTAPP:
+                BrowserStore.setVaultPreferenceToMattermostApp(this.props.siteUrl);
+                break;
+            case VaultPreferenceTypes.BROWSER:
+                BrowserStore.setVaultPreferenceToBrowser(this.props.siteUrl);
+                break;
+            default:
+                break;
+            }
+        }
+    }
+
+    openMattermostApp = () => {
+        this.setPreference(VaultPreferenceTypes.MATTERMOSTAPP);
+        this.setState({redirectPage: true});
+        window.location.href = this.props.nativeLocation;
+    }
+
+    openInBrowser = () => {
+        this.setPreference(VaultPreferenceTypes.BROWSER);
+        window.location.href = this.props.location;
+    }
+
+    tryOpen = () => {
+        safeOpenProtocol(this.props.nativeLocation,
+            () => this.setState({protocolUnsupported: true}),
+            () => protocolDetected(),
+            () => this.setState({browserUnsupported: true})
+        );
+    }
+
+    render() {
```

Comment:
This is a pretty long render function, might be worth breaking up into smaller ones